### PR TITLE
[GOBBLIN-897] adds local FS spec executor to write jobs to a local dir

### PR DESCRIPTION
**Pull request by Will-Lo**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-897

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):

Writes JobSpecs to a local folder, where presumably an instance of Gobblin-standalone can pick up the job to execute them. This is primarily designed for onboarding new users of GaaS.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by Will-Lo** [comment]
@sv2000  @zxcware  could you take a look? Thanks!

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_executorInstance/LocalFsSpecProducer.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+      throw new RuntimeException("Unsupported spec type " + spec.getClass());
+    }
+  }
+
+  /** Delete a {@link Spec} being executed on {@link org.apache.gobblin.runtime.api.SpecExecutor}.
+   * @param deletedSpecURI
+   * @param headers*/
+  @Override
+  public Future<?> deleteSpec(URI deletedSpecURI, Properties headers) {
+    return new CompletedFuture<>(Boolean.TRUE, null);
```
Comment (zxcware) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_executorInstance/LocalFsSpecProducer.java]: are we expected to delete the job corresponds to the uri?

Comment (Will-Lo) [reply gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_executorInstance/LocalFsSpecProducer.java]: Ah oops I should implement that, good point

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_executorInstance/LocalFsSpecProducer.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  @Override
+  public Future<?> updateSpec(Spec updatedSpec) {
+    return writeSpec(updatedSpec, SpecExecutor.Verb.UPDATE);
+  }
+
+  private Future<?> writeSpec(Spec spec, SpecExecutor.Verb verb) {
+    if (spec instanceof JobSpec) {
+      String specString = spec.toString();
+      // format the JobSpec to have file of <flowGroup>_<flowName>.job
+      String jobFileName = specString.replace('/', '_').substring(specString.lastIndexOf(':')+2, specString.length()-1) + ".job";
```
[the remark this answers is not in the document]
Comment (Will-Lo) [reply gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_executorInstance/LocalFsSpecProducer.java]: By stripping out the scheme, would I be using `uri.getPath()`? Thanks for the insight though! Appreciated.

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_executorInstance/LocalFsSpecProducer.java [hunk 3]
```diff
@@ -71,9 +72,10 @@ public LocalFsSpecProducer(Config config) {
 
   private Future<?> writeSpec(Spec spec, SpecExecutor.Verb verb) {
     if (spec instanceof JobSpec) {
-      String specString = spec.toString();
+      URI specUri = spec.getUri();
       // format the JobSpec to have file of <flowGroup>_<flowName>.job
-      String jobFileName = specString.replace('/', '_').substring(specString.lastIndexOf(':')+2, specString.length()-1) + ".job";
+      String[] uriTokens = specUri.getPath().split("/");
```
Comment (zxcware) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_executorInstance/LocalFsSpecProducer.java]: We can be more DRY(Don't Repeat Yourself) by making a small function named `getJobFileName(uri)`

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_executorInstance/LocalFsSpecProducer.java [hunk 4]
```diff
@@ -94,7 +96,14 @@ public LocalFsSpecProducer(Config config) {
...
   public Future<?> deleteSpec(URI deletedSpecURI, Properties headers) {
-    return new CompletedFuture<>(Boolean.TRUE, null);
+    String[] uriTokens = deletedSpecURI.getPath().split("/");
+    String jobFileName = String.join("_", uriTokens) + ".job";
+    File file = new File(jobFileName);
+    if (file.delete()) {
+      log.info("Deleted spec: {}", jobFileName);
+      return new CompletedFuture<>(Boolean.TRUE, null);
+    }
+    throw new RuntimeException(String.format("Failed to delete file with uri %s", deletedSpecURI));
```
Comment (zxcware) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_executorInstance/LocalFsSpecProducer.java]: This shouldn't halt the service, return a `CompletedFuture` with an exception and expect the invoker to do proper error handling.

Comment (Will-Lo) [reply gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_executorInstance/LocalFsSpecProducer.java]: That's a good point actually. In the other `SpecProducers` they throw a runtime exception as well, would it affect anything?

**Pull request closed without merging**